src/types.ts

```diff
@@ -0,0 +1,13 @@
+import { type MapeoValue as AllMapeoValues } from './schema/index.js'
+export type SupportedSchemaNames = 'preset' | 'field' | 'observation'
+
+/** Filter a union of objects to only include those that have a prop `schemaName` that matches U */
+export type FilterBySchemaName<
+  T extends { schemaName: string },
+  U extends string,
+> = Extract<T, { schemaName: U }>
+
+export type MapeoValue = FilterBySchemaName<
```

Comment:
We can't satisfy this constraint since no schema has a `schemaType` field (observation has `type` field). So I don't know how to type the object passed for validation

Reply:
If we have certainty that we are only migrating Observations, then could be very much simplified